massa-execution/src/worker.rs

```diff
@@ -255,106 +253,137 @@ impl ExecutionWorker {
+        self.pending_css_final_blocks
+            .extend(finalized_blocks.into_iter().filter_map(|(b_id, b)| {
+                if b.header.content.slot <= self.last_active_slot {
+                    // eliminate blocks that are not from a stricly later slot than the current latest SCE-final one
+                    return None;
+                }
+                Some((b.header.content.slot, (b_id, b)))
+            }));
+        if let Some(max_css_final_slot) = self
+            .pending_css_final_blocks
+            .last_key_value()
+            .map(|(s, _v)| *s)
+        {
+            let mut cur_slot = self.last_final_slot.get_next_slot(self.cfg.thread_count)?;
+            while cur_slot <= max_css_final_slot {
+                match self
+                    .pending_css_final_blocks
+                    .first_key_value()
+                    .map(|(s, _v)| *s)
+                {
+                    // there is a CSS-final block at cur_slot
+                    Some(b_slot) if b_slot == cur_slot => {
+                        // remove the entry from pending_css_final_blocks (cannot panic, checked above)
+                        let (_b_slot, (b_id, block)) =
+                            self.pending_css_final_blocks.pop_first().unwrap();
+                        // execute the block as a SCE-final VM step
+                        self.push_request(ExecutionRequest::RunFinalStep(ExecutionStep {
+                            slot: cur_slot,
+                            block: Some((b_id, block)),
+                        }));
+                        // update cursors
+                        self.last_active_slot = cur_slot;
+                        self.last_final_slot = cur_slot;
+                    }
 
-        // list maximum thread slots
-        let mut max_thread_slot = vec![self.last_final_slot; self.cfg.thread_count as usize];
-        for (_b_id, block) in css_final_blocks.iter() {
-            max_thread_slot[block.header.content.slot.thread as usize] = std::cmp::max(
-                max_thread_slot[block.header.content.slot.thread as usize],
-                block.header.content.slot,
-            );
-        }
+                    // there is no CSS-final block at cur_slot, but there are CSS-final blocks later
+                    Some(_b_slot) => {
+                        // check whether there is a CSS-final block later in the same thread
+                        let mut check_slot = Slot::new(cur_slot.period + 1, cur_slot.thread);
+                        while check_slot <= max_css_final_slot {
+                            if self.pending_css_final_blocks.contains_key(&check_slot) {
+                                break;
+                            }
+                            check_slot.period += 1;
+                        }
+                        if check_slot <= max_css_final_slot {
+                            // subsequent CSS-final block found in the same thread as cur_slot
```

Comment:
`cur_slot` should be renamed, or clearly commented, as `slot_s`, or something similar, to match the spec(or the spec should use different wording).

Reply:
done